tdrs-backend/tdpservice/parsers/util.py

```diff
@@ -80,12 +122,12 @@ def get_all_fields(self):
         """Get all fields from the schema."""
         return self.fields
 
-    def parse_and_validate(self, line):
+    def parse_and_validate(self, line, generate_error):
```

Comment:
Instead of passing this into every function that needs it, why not just use `from x import y as generate_error`?

Reply:
Passing it in allows us to use a different `generate_error` in different places of the code (for different schemas or operations, etc). One use case might be the trailer, where maybe we want to log an error but not generate a `ParserError`, the implementation can be easily changed for the one schema by passing in a different function.